src/resolvers/User/post.ts

```diff
@@ -0,0 +1,96 @@
+import { Post } from "../../models";
+import type { UserResolvers } from "../../types/generatedGraphQLTypes";
+/**
+ * This resolver function will fetch and return the post created by the user from database.
+ * @param parent - An object that is the return value of the resolver for this field's parent.
+ * @returns An array which conatins the post created by the user.
+ */
+
+export const posts: UserResolvers["posts"] = async (parent, args) => {
+  const pageSize: number =
```

Comment:
stop making use of `as` keyword, it's an escape hatch to circumvent type safety, it should be used sparingly

look at this https://github.com/PalisadoesFoundation/talawa-api/blob/main/src/utilities/graphqlConnectionFactory.ts

the logic is a bit different but the implementation is correct, create a generic connection argument validation function which validates that:-
1. either `first` or `last` are provided
2. when `first` is provided, `last` isn't provided
3. either `first` and `after` or `last` and `before` are provided
4. when `first` and `after` are provided, `last` and `before` aren't provided